### DISCUSSION TITLE
Code improvements (Avoiding copying)

### DISCUSSION THREAD

**Discussion by MuaazH** [T-Troll/alienfx-tools#293] [Ideas]
So... Afx_lightblock is a bit too heavy
``` C++
	struct Afx_lightblock { // light action block
		byte index;
		vector<Afx_action> act;
	};
```
as you can see, you have to add `Afx_action` structures to `Afx_lightblock.act`, this makes copies for them, when it really does not have to.
What we could do is change the structure to:
``` C++
	struct Afx_lightblock { // light action block
		byte index;
		vector<Afx_action *> act;
	};
```
Or, we could just not use std::vectors:
``` C++
	struct Afx_lightblock { // light action block
		byte index; // alginment will waste 3 bytes on x32 systems, so we might as well upgrade this to int
                int count;
		Afx_action * act;
	};
```

This should should cause no problems, since the code is not meant to be thread safe anyway ANNND the library code does not need the data after returning. The caller can clean up thier own memory after the call.

If council of contributors agree, then I shall make the changes with haste. Probably.

**Comment by T-Troll** [comment]
Not a good idea, in fact.
Main code is multithreaded, so this will break vector pointer in case one thread modify an action vector/array (it can be reallocated).
Copy constructors much more stable.
